sonar-html-plugin/src/main/java/org/sonar/plugins/html/checks/sonar/LangAttributeCheck.java

```diff
@@ -39,15 +39,31 @@ private static boolean isHtmlTag(TagNode node) {
   private static boolean hasLangAttribute(TagNode node) {
     return node.hasProperty("lang")
       || node.hasProperty("xml:lang")
-      || hasLangAttributeFromWordPress(node);
+      || hasWordPressLangAttribute(node)
+      || hasThymeleafLangAttribute(node);
   }
 
   /**
    * Using WordPress, HTML attributes can be set using the php function `language_attributes`
    */
-  private static boolean hasLangAttributeFromWordPress(TagNode node) {
+  private static boolean hasWordPressLangAttribute(TagNode node) {
     return node.getAttributes().stream()
       .map(Attribute::getName)
       .anyMatch(attributeName -> attributeName.contains("?php") && attributeName.contains("language_attributes"));
   }
+
+  /**
+   * In Thymeleaf there are multiple ways of specifying the lang attribute:
+   * - using the th:lang, th:xmllang, th:lang-xmllang attributes (lang-xmllang would set both xmllang and lang attributes)
+   * - using the th:attr attribute for specifying different attributes. Example "th:attr="lang=html,xml:lang=html""
+   */
+  private static boolean hasThymeleafLangAttribute(TagNode node) {
+    String thAttrValue = node.getAttribute("th:attr");
+    return node.hasProperty("th:lang")
+      || node.hasProperty("th:xmllang")
+      || node.hasProperty("th:lang-xmllang")
+      || (thAttrValue != null && thAttrValue.contains("lang")
```

Comment:
To reduce the possibility of False Negatives, it might be nicer to check for `thAttrValue.contains("lang=")` instead, otherwise if the `lang` string appears anywhere in the attribute, it will be considered valid. Example: `<html th:attr="class="my-lang-class"" >` would be wrongly considered as compliant.
False Negatives would still be possible with the check for `lang=`, (for instance in the case of `<html th:attr="title="lang=en"" >`.) but would be less likely.

Reply:
Sounds good !